src/services/fetch.service.ts

```diff
@@ -0,0 +1,37 @@
+class FetchService {
+  private responseMap = new Map<string, Promise<Response>>();
+
+  private async fetchData(endpoint: string, init?: RequestInit): Promise<Response> {
+    const url = this.getUrl(endpoint);
+
+    if (this.responseMap.has(url)) return this.responseMap.get(url)!;
+
+    const fetchPromise = fetch(url, init);
+
+    this.responseMap.set(url, fetchPromise);
+
+    const response = await fetchPromise;
+
+    if (response.ok) return response;
+
+    console.error(response.statusText);
+    return response;
+  }
+
+  public async fetchJson<T>(endpoint: string, init?: RequestInit): Promise<T> {
+    const response = await this.fetchData(endpoint, init);
+    return await response.clone().json();
+  }
+
+  public async fetchText(endpoint: string, init?: RequestInit): Promise<string> {
+    const response = await this.fetchData(endpoint, init);
+    return response.clone().text();
```

Comment:
wieso mit clone() ?

Reply:
Clone wird benötigt damit man eine response mehrmals auflösen kann.

Ansonsten wird der Fehler geworfen : TypeError: Failed to execute 'json' on 'Response': body stream already read

MDN
The clone() method of the Response interface creates a clone of a response object, identical in every way, but stored in a different variable. The main reason clone() exists is to allow multiple uses of Body objects (when they are one-use only.)

https://developer.mozilla.org/en-US/docs/Web/API/Response/clone